src/server/hotkey_collector.cpp

```diff
@@ -49,5 +66,50 @@ void hotkey_collector::capture_hash_key(const dsn::blob &hash_key, int64_t weigh
 
 void hotkey_collector::analyse_data() { _internal_collector->analyse_data(); }
 
+void hotkey_collector::on_start_detect(dsn::replication::detect_hotkey_response &resp)
+{
+    auto now_state = _state.load();
+    std::string hint;
+    switch (now_state) {
+    case hotkey_collector_state::COARSE_DETECTING:
+    case hotkey_collector_state::FINE_DETECTING:
+        resp.err = dsn::ERR_SERVICE_ALREADY_EXIST;
+        hint = fmt::format("still detecting {} hotkey, state is {}",
+                           dsn::enum_to_string(_hotkey_type),
+                           enum_to_string(now_state));
+        ddebug_replica(hint);
+        return;
+    case hotkey_collector_state::FINISHED:
+        resp.err = dsn::ERR_SERVICE_ALREADY_EXIST;
+        hint = fmt::format(
+            "{} hotkey result has been found, you can send a stop rpc to restart hotkey detection",
+            dsn::enum_to_string(_hotkey_type));
+        ddebug_replica(hint);
+        return;
+    case hotkey_collector_state::STOPPED:
+        // TODO: (Tangyanzhao) start coarse detecting
+        _state.store(hotkey_collector_state::COARSE_DETECTING);
+        resp.err = dsn::ERR_OK;
+        hint = fmt::format("starting to detect {} hotkey", dsn::enum_to_string(_hotkey_type));
+        ddebug_replica(hint);
+        return;
+    default:
+        hint = "invalid collector state";
+        resp.err = dsn::ERR_INVALID_VERSION;
+        resp.__set_err_hint(hint);
+        dwarn_replica(hint);
```

Comment:
Better to assert here, coz the state is internal enum, default case will not happen.